torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -782,6 +831,92 @@ void InsertQuantDeQuantHelper::extractAndRunWeightObserver(
   weight_to_graph_fn_[weight_value]->run(module_inp);
 }
 
+Node* insertEmbeddingBagOps(
+    Module& module,
+    Node* observer,
+    const std::string& op_name) {
+  Graph* g = observer->owningGraph();
+  auto observer_out = observer->output();
+
+  std::string prepack_fn, quant_fn;
+  if (op_name == "embedding_bag_4bit") {
+    prepack_fn = "quantized::embedding_bag_4bit_prepack";
+    quant_fn = "quantized::embedding_bag_4bit_rowwise_offsets";
+  } else if (op_name == "embedding_bag_byte") {
+    prepack_fn = "quantized::embedding_bag_byte_prepack";
+    quant_fn = "quantized::embedding_bag_byte_rowwise_offsets";
+  } else {
+    TORCH_INTERNAL_ASSERT(
+        "Graph Mode Quantization currently supports 4-bit and 8-bit embedding bag quantization.");
+  }
+
+  std::vector<Value*> prepack_inputs = {observer_out};
+  std::vector<Use> uses = observer_out->uses();
+  Node* embedding_bag_float_op;
+  for (const Use& use : uses) {
+    if (matchCallFuncToUse(use, "embedding_bag", 2)) {
+      embedding_bag_float_op = use.user;
+    }
+  }
```

Comment:
what are the possible cases here? can we have stronger guarantees here? is the observer_out always going to be used by an embedding_bag op here?